File: cpp/3. Basic math/3_palindrome.cpp
#include<bits/stdc++.h>
using namespace std;
bool palindrome(int x)
{
    int copy=x;
    int rev=0;
    while (x>0)
    {
        int ld=x%10;
        rev=(10*rev)+ld;
        x/=10;
    }
    if (rev==copy)
    return true;
    else
    return false;

    
    
    
}
int main()
{
     int n;
  cout<<"Enter number:";
  cin>>n;
  bool ans=palindrome(n);
  if(ans)
  cout<<n<<" is palindrome";
  else
  {
    cout<<n<<" is not palindrome";
  }
  
    
    return 0;
}

File: python/2. Basic Math/3_palindrome.py
def palindrome(n):
    dup=n
    revNum=0
    while n>0:
        ld=n%10
        revNum=(10*revNum)+ld
        n=n//10
    if dup==revNum:
        return True
    else:
        return False
def main():
    number=121
    if palindrome(number):
        print(number, " is a palindrome")
    else:
        print(number, " is not a palindrome")

if __name__ =="__main__":
    main()
